src/lib/fetch.ts

```diff
@@ -9,85 +10,99 @@ export interface FetchOptions {
   noResolveJson?: boolean
 }
 
-export type RequestMethodType = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE'
+export interface FetchParameters {
+  signal?: AbortSignal
+}
+
+export type RequestMethodType = 'GET' | 'POST' | 'PUT' | 'DELETE'
 
 const _getErrorMessage = (err: any): string =>
   err.msg || err.message || err.error_description || err.error || JSON.stringify(err)
 
-const handleError = (error: any, reject: any) => {
-  if (!error?.status) {
-    return reject({ message: NETWORK_FAILURE.ERROR_MESSAGE })
-  }
-  if (typeof error.json !== 'function') {
-    return reject(error)
-  }
-  error.json().then((err: any) => {
-    return reject({
-      message: _getErrorMessage(err),
-      status: error?.status || 500,
+const handleError = (error: unknown, reject: (reason?: any) => void) => {
+  const Res = resolveResponse()
+
+  if (error instanceof Res) {
+    error.json().then((err) => {
+      reject(new AuthError(_getErrorMessage(err), error.status || 500))
     })
-  })
+  } else {
+    reject(new AuthUnknownError(_getErrorMessage(error), error))
+  }
 }
 
-const _getRequestParams = (method: RequestMethodType, options?: FetchOptions, body?: object) => {
+const _getRequestParams = (
+  method: RequestMethodType,
+  options?: FetchOptions,
+  parameters?: FetchParameters,
+  body?: object
+) => {
   const params: { [k: string]: any } = { method, headers: options?.headers || {} }
 
   if (method === 'GET') {
     return params
   }
```

Comment:
hmm why don't we return `parameters` if it's a get request?

Reply:
Seems to be in the prior implementation as well. I used the fetch code from the storage client lib.